app/supplejack/extraction/javascript_request.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Extraction
+  class JavascriptRequest
+    def initialize(url:, params:)
+      @url = url
+      @params = params
+      @driver = driver
+
+      @document
+    end
+
+    def get
+      begin
+        @driver.get(full_url)
+        @document = document(200)
+      rescue
+        @document = document(500)
+      end
+      
+      # Quit after assigning the document so that the browser process is stopped
+      @driver.quit
```

Comment:
I wonder if this could be in a `ensure` block

https://ruby-doc.org/docs/ruby-doc-bundle/UsersGuide/rg/ensure.html

> The ensure clause does just this. ensure goes after the last rescue clause and contains a chunk of code that will always be executed as the block terminates. It doesn't matter if the block exits normally, if it raises and rescues an exception, or if it is terminated by an uncaught exception---the ensure block will get run.

Reply:
That sounds good! I will give it a go :)